src/Feature.php

```diff
@@ -0,0 +1,74 @@
+<?php
+
+namespace P4\MasterTheme;
+
+abstract class Feature {
+
+	public const OPTIONS_KEY = 'planet4_features';
+
+	/**
+	 * Default to using the class name, but allow override to provide BC.
+	 *
+	 * @return string
+	 */
+	abstract public static function get_id(): string;
+//	public static function get_id(): string {
+//		return preg_replace( '/.*\/', '', self::class );
+//	}
+
+	/**
+	 * Get the translated name. It's important the class includes the translation function call. This is used by
+	 * automation to extract translatable strings from the source code.
+	 *
+	 * @return string A human readable name.
+	 */
+	abstract public static function get_name(): string;
+
+	/**
+	 * Get the translated description. It's important the class includes the translation function call. This is used by
+	 * automation to extract translatable strings from the source code.
+	 *
+	 * @return string
+	 */
+	abstract public static function get_description(): string;
+
+	/**
+	 * @return bool Whether the feature toggle can be used on production.
+	 */
+	abstract public static function is_live(): bool;
+
+	/**
+	 * Not added as abstract as this function will only return true when we remove it.
+	 *
+	 * @return bool Whether the feature has become generally available, meaning it's always on for everyone.
+	 */
+	public static function is_generally_available(): bool {
+		return false;
```

Comment:
After adding this I'm unsure if it makes much sense. If we on a code level set it up so that `is_active` always returns `true`, it's effectively the same as removing the old code, as it becomes unreachable.

Then again, it's easy to revert this one line change. So while it also needs a redeploy to re-enable the feature, there's no chance for conflicts if we'd have to restore the code.

Reply:
The way we do this at the moment is
- Run a patch to activate the feature everywhere
- When ready, remove the old code and the feature

The issue I'd have with this `is_generally_available` method is I'm not sure it would be reflected on the feature page; is the checkbox checked ? Does the user see the activated feature activated and locked ?

Reply:
I didn't further develop the behavior as no features would use it at this moment. If we'd use this then what you say makes sense to do :+1: 

Though still not sure whether this function makes sense. If we want to remove a feature, we should have already switched all sites. And it's relatively easy to confirm that an option is indeed switched on all sites. We could even have a job on `planet4-base`, or in each site, to enforce this.